woocommerce/woocommerce-admin
# Show keep & dismiss page when tasks are complete

Fixes #7634 

This PR adds the following changes.

1. Adds a modal when the user tries to hide the task list when there is at least one incomplete task.
2. Prompt the user with two buttons either to keep/hide the task list when all the tasks are complete.

### Screenshots
<img width="825" alt="image" src="https://user-images.githubusercontent.com/5121465/133145744-d641e81f-de83-435c-8464-6d9d8e306de0.png">

![image](https://user-images.githubusercontent.com/9312929/132516484-6633a1cc-bd67-42d1-8261-735a2e5331c8.png)

### Detailed test instructions:

**Case 1: Modal should popup when there is at least one incomplete task.**

1. Start with a fresh installation.
2. Onboard as usual.
3. Click the ellipses menu and choose `Hide task list`
4. The modal should pop up. Click `Yes, hide store setup tasks` and confirm the task list is no longer visible.

**Case 2: Modal should not popup when all the tasks are complete**

1. Start with a fresh installation.
2. Onboard as usual.
3. Click the ellipses menu and choose `Hide task list`
4. The task list should be invisible without prompting the modal.

**Case 3: You've completed store setup should be displayed when all the tasks are complete**

1. Start with a fresh installation.
2. Onboard as usual.
3. Finish all the available tasks.
4. You should see [ Keep list ] and [ Hide this list ] with `You've completed store setup` heading.
5. Click [ Keep list ] and confirm the task list is visible again.
6. Refresh the page and confirm the task list is still visible.

**Case 4: Clicking [ Hide this list ] button should hide the task list**

1. Start with a fresh installation.
2. Onboard as usual.
3. Finish all the available tasks.
4. You should see [ Keep list ] and [ Hide this list ] with `You've completed store setup` heading.
5. Click [ Hide this list ] and confirm the task list is hidden.
6. Refresh the page and confirm the task list is still hidden.
